CustomerManagementPanel/src/app/addresses/addresses-add/addresses-add.component.ts

```diff
@@ -1,48 +1,117 @@
 import { FormControl, FormGroup, Validators } from '@angular/forms';
-import { Address, AddressForCreation } from 'src/app/clients/models/address';
+import { ActivatedRoute } from '@angular/router';
+import { faSave, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { ToastrService } from 'ngx-toastr';
+import { Observable, of } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+import {
+  Address,
+  AddressForCreation,
+  Country,
+} from 'src/app/clients/models/address';
 import { MyFormGroup } from 'src/app/shared/extentions/myFormGroup';
 import { AddressesService } from '../addresses.service';
 
-
 @Component({
   selector: 'app-addresses-add',
   templateUrl: './addresses-add.component.html',
-  styleUrls: ['./addresses-add.component.scss']
+  styleUrls: ['./addresses-add.component.scss'],
 })
 export class AddressesAddComponent implements OnInit {
-
-  address: AddressForCreation = { country: "", city: "", street: "", zipCode: ""};
-  form: MyFormGroup = null; 
+  @Input() addressesList: Address[];
+  copyAddressList: AddressForCreation[] = [];
+  address: AddressForCreation = {
+    countryId: '',
+    city: '',
+    street: '',
+    zipCode: '',
+  };
+  form: MyFormGroup = null;
+  faSave = faSave;
+  faTimes = faTimes;
+  baseUrl: string = 'https://localhost:44391/api/addresses';
+  addressForCreation: AddressForCreation[] = [];
+  countries: Country[] = [];
+  polandId: string;
 
   constructor(
+    private httpClient: HttpClient,
+    private toastr: ToastrService,
     private addressService: AddressesService,
+    private route: ActivatedRoute
   ) {
-    this.form = new MyFormGroup({
-    "country": new FormControl(this.address.country,[Validators.required, Validators.minLength(3), Validators.maxLength(30), Validators.pattern('[a-zA-Z ]*')]),
-    "zipCode": new FormControl(this.address.zipCode,[Validators.required, Validators.minLength(5), Validators.maxLength(6), Validators.pattern('[0-9]*')]),
-    "city": new FormControl(this.address.city,[Validators.required, Validators.minLength(3), Validators.maxLength(30), Validators.pattern('[a-zA-Z ]*')]),
-    "street": new FormControl(this.address.street,[Validators.required, Validators.minLength(3), Validators.maxLength(40)])
-  }); }
+    this.addressService.getAllCountries().subscribe((response) => {
+      this.countries = response;
+
+      this.polandId = this.countries.find((x) => x.name === 'Poland')?.id;
 
+      let initialCountry =
+        this.address.countryId === '' || !this.address.countryId
+          ? this.polandId
+          : this.address.countryId;
+
+    this.form = new MyFormGroup({
+      countryId: new FormControl(initialCountry, [
+        Validators.required]),
+      zipCode: new FormControl(this.address.zipCode, [
+        Validators.required,
+        Validators.minLength(5),
+        Validators.maxLength(7),
+        Validators.pattern('[0-9]{2}\-[0-9]{3}'),
```

Comment:
Magic pattern, please put it into a properly named variable first and then use it in this Validators.pattern.

Reply:
let zipCodePattern = '[0-9]{2}\-[0-9]{3}' // XX-XXX
let mailPattern = "....." // xxxxx@xx.xx